src/main/java/mate/jdbc/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,133 @@
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't can't insert manufacturer "
+                    + manufacturer.getName() + " to DB", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        Optional<Manufacturer> result = Optional.empty();
+        String getManufacturerQuery = "SELECT * FROM manufacturers "
+                + "WHERE is_deleted = false AND id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement =
+                        connection.prepareStatement(getManufacturerQuery)) {
+            getManufacturerStatement.setString(1, String.valueOf(id));
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            if (resultSet.next()) {
+                result = Optional.of(parseResultToManufacturer(resultSet));
+            }
+            return result;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String getAllManufacturersQuery = "SELECT * FROM manufacturers WHERE is_deleted = false";
+        List<Manufacturer> result = new LinkedList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement =
+                        connection.prepareStatement(getAllManufacturersQuery)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                result.add(parseResultToManufacturer(resultSet));
+            }
+            return result;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufactures", e);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufacturerQuery = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE is_deleted = false AND id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturerStatement =
+                        connection.prepareStatement(updateManufacturerQuery)) {
+            setStatements(updateManufacturerStatement, manufacturer);
+            updateManufacturerStatement.setLong(3, manufacturer.getId());
+            int executeUpdate = updateManufacturerStatement.executeUpdate();
+            if (executeUpdate > 0) {
+                return manufacturer;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer by id: "
+                    + manufacturer.getId(), e);

```

Comment:
put all info about manufacturer 
```suggestion
            throw new DataProcessingException("Can't update manufacturer: "
                    + manufacturer, e);

```